pkg/services/sqlstore/migrations/ualert/ualert.go

```diff
@@ -288,6 +352,7 @@ func (m *migration) writeAlertmanagerConfig(amConfig *PostableUserConfig, allCha
 		// Since we are migration for a snapshot of the code, it is always going to migrate to
```

Comment:
To answer the above TODO, we shouldn't because the config will be applied on service startup and before it is ready. My understanding is that migrations are run before we start up services.

Reply:
I don't fully understand this TODO but I confirm that the migrations run first (the `sqlstore` service that starts the migrations runs with [higher](https://github.com/grafana/grafana/blob/11d03f7961db3cd7dd613d94a843d1a40f270b19/pkg/services/sqlstore/sqlstore.go#L71) priority)